src/bugsnag/client.ts

```diff
@@ -1067,6 +1083,456 @@ export class BugsnagClient implements Client {
+        const result = await this.projectApi.listProjectSpanGroups(
+          project.id,
+          params.sort,
+          params.direction,
+          params.perPage,
+          undefined,
+          params.filters as Array<PerformanceFilter> | undefined,
+          params.starredOnly,
+          params.nextUrl,
+        );
+        return {
+          content: [
+            {
+              type: "text",
+              text: JSON.stringify({
+                data: result.body,
+                next_url: result.nextUrl,
+                count: result.body?.length,
+              }),
+            },
+          ],
+        };
+      },
+    );
+
+    const getSpanGroupInputSchema = z.object({
+      projectId: this.projectApiKey
+        ? toolInputParameters.projectId.optional()
+        : toolInputParameters.projectId,
+      spanGroupId: z
+        .string()
+        .describe("ID of the span group (will be URL-encoded automatically)"),
+      filters: PerformanceFiltersArraySchema.optional().describe(
+        "Apply filters to narrow down the span group list. Use the List Trace Fields tool to discover available filter fields",
+      ),
+    });
+
+    register(
+      {
+        title: "Get Span Group",
+        summary: "Get detailed performance metrics for a specific span group",
+        purpose: "Analyze performance characteristics of a specific operation",
+        useCases: [
+          "View detailed statistics (p50, p75, p90, p95, p99) for an operation",
+          "Check if performance targets are configured",
+          "Monitor span count to understand operation volume",
+        ],
+        inputSchema: getSpanGroupInputSchema,
+        examples: [
+          {
+            description: "Get details for an API endpoint span group",
+            parameters: { spanGroupId: "[HttpClient]GET-api.example.com" },
+            expectedOutput: "Statistics, category, and performance target info",
+          },
+          {
+            description: "Get span group details with device filtering",
+            parameters: {
+              spanGroupId: "[HttpClient]GET-api.example.com",
+              filters: [
+                {
+                  key: "device.browser_name",
+                  filterValues: [{ matchType: "eq", value: "Chrome" }],
+                },
+              ],
+            },
+            expectedOutput: "Statistics filtered for Chrome browser only",
+          },
+        ],
+        hints: [
+          "Use List Span Groups first to discover available span group IDs",
+          "IDs are automatically URL-encoded - provide the raw ID",
+          "Statistics include p50, p75, p90, p95, p99 percentiles",
+        ],
+      },
+      async (args, _extra) => {
+        const params = getSpanGroupInputSchema.parse(args);
+        const project = await this.getInputProject(params.projectId);
+        if (!params.spanGroupId) {
+          throw new ToolError("spanGroupId is required");
+        }
```

Comment:
No need to validate this – it's done in the schema `parse`

Also in the other tools